components/scream/CMakeLists.txt

```diff
@@ -401,15 +414,21 @@ include(EkatMpiUtils)
 # (for openpmi; -lmpicxx for mpich) flag.
 DisableMpiCxxBindings()
 
+message("CMAKE_CXX_COMPILER_ID is ${CMAKE_CXX_COMPILER_ID}")
 if (SCREAM_DOUBLE_PRECISION)
   if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
     set(SCREAM_Fortran_FLAGS -real-size 64)
+  elseif("${CMAKE_CXX_COMPILER_ID}" MATCHES "[Cc]lang" OR "${CMAKE_CXX_COMPILER_ID}" MATCHES "HIPCC")
+    #find what is crayclang id in cmake
+    set(SCREAM_Fortran_FLAGS -s default32 -eZ)
```

Comment:
Would you put a comment explaining what these do or why they are here? Does -default32 disagree with -fdefault-real-8 below?

Reply:
I did not investigate this, i think early on i copied these flags from a Xingqiu branch. so i cannot provide a coerent comment, maybe, @xyuan could? thanks